client/tx/query.go

```diff
@@ -62,21 +57,32 @@ func queryTx(cdc *wire.Codec, cliCtx context.CLIContext, hashHexStr string, trus
 		return nil, err
 	}
 
-	res, err := node.Tx(hash, !trustNode)
+	res, err := node.Tx(hash, !cliCtx.TrustNode)
 	if err != nil {
 		return nil, err
 	}
 
-	info, err := formatTxResult(cdc, res)
+	info, err := formatTxResult(cdc, cliCtx, res)
 	if err != nil {
 		return nil, err
 	}
 
 	return wire.MarshalJSONIndent(cdc, info)
 }
 
-func formatTxResult(cdc *wire.Codec, res *ctypes.ResultTx) (Info, error) {
-	// TODO: verify the proof if requested
+func formatTxResult(cdc *wire.Codec, cliCtx context.CLIContext, res *ctypes.ResultTx) (Info, error) {
+	if !cliCtx.TrustNode {
```

Comment:
This logic is fine but I think it's confusing to put it in a function called `formatTxResult` - at first I thought you deleted the proof verification because I expected `formatTxResult` to only do formatting! Can we separate this into a different function - maybe `validateTxResult` or something?

Reply:
I added the verification logic here because I noticed the TODO. 
How about I create a new function named `validateTxResult` and call it in `formatTxResult`, like this:
```
func formatTxResult(cdc *wire.Codec, cliCtx context.CLIContext, res *ctypes.ResultTx) (Info, error) {
     if !cliCtx.TrustNode {
          if err := validateTxResult(cliCtx , res); err !=nil {
                 return ni, err
          }
     }
     ....
}
```

Reply:
I'm sorry for misunderstanding. I have changed the code move the tx verification out of tx format.